pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -95,6 +95,7 @@ public class PersistentDispatcherMultipleConsumers extends AbstractDispatcherMul
     protected boolean shouldRewindBeforeReadingOrReplaying = false;
     protected final String name;
     protected boolean sendInProgress;
+    protected boolean sendInProgressReplay;
```

Comment:
Maybe we can combine the two variables into a counter? Just like this:

```java
AtomicInteger sendingTaskCounter;

sendingTaskCounter.increment();
try {
  trySendMessagesToConsumer();
} finally {
  int runningTaskCount = sendingTaskCounter.decrementAndGet();
  if (runningTaskCount == 0){
    readMoreEntries();
  }
}
```

I think this makes the logic easier to unstand.

Reply:
Good idea, but seems to need to remove `sendInProgress=false` in the PersistentStickyKeyDispatcherMultipleConsumers, I use delay execute `readMoreEntries` to replace it.